sros2/sros2/api/__init__.py

```diff
@@ -294,38 +334,39 @@ def create_key(keystore_path, identity):
                 print('Existing symlink does not match!')
                 raise RuntimeError(str(e))
 
-    # copy the governance file in there
-    keystore_governance_path = os.path.join(keystore_path, 'governance.p7s')
+    # symlink the governance file in there
+    keystore_governance_path = os.path.join(keystore_path, KS_CONTEXT, 'governance.p7s')
     dest_governance_path = os.path.join(key_dir, 'governance.p7s')
-    shutil.copyfile(keystore_governance_path, dest_governance_path)
+    relativepath = os.path.relpath(keystore_governance_path, key_dir)
+    os.symlink(src=relativepath, dst=dest_governance_path)
```

Comment:
What happens in the destination file already exists?

Reply:
If the file already exists, it will raise a `FileExistsError`.
Should the old file be deleted and overridden?

Reply:
Addressed in https://github.com/ros2/sros2/pull/177/commits/f5080e8bfa4ea0ec88d5db87d7dd23ba841b9fb1.

Instead of `delete + override`, I'm raising an error in that last case.
I'm afraid to delete something a distracted user didn't want.

Reply:
Maybe it's my workflow but I find it convenient to change a couple rules in a policy file and just rerun `generate_artifacts` on my entire policy file and trust the tool to override the changed permission files accordingly. It looks like with this more conservative approach I would need to go prune by hand all the directories that may have a permission change.

I'd prefer not changing the previous behavior, but either way it would be good to provide an option to opt-in (or out) of overwriting existing files

Reply:
In my prior work with keymint, I treated auto generated security files as build artifacts, and thus subsequent build file could be modified at will by the build tool, where the user would recognize the temporal nature of these files if they instructed the build tool to rebuild changed source files. So I'd say let the tool overwrite it the generated files by default.

In keymint, I also separated the pipeline into a src. build, and install folder structure, so we may want to explore using similar workflows that could separate the policy source files, from the generated intermediate representations built, to the final signed security artifacts installed. Keymint had a `--dry-run` flag that the CLI could check to see if it should warn about detected file diffs and not write to disk.

Reply:
Done in https://github.com/ros2/sros2/pull/177/commits/d52333215652ccce161473dbf722a3d4a40c5300.